lib/reek/cli/option_interpreter.rb

```diff
@@ -0,0 +1,82 @@
+require 'forwardable'
+require 'reek/cli/input'
+require 'reek/cli/report/report'
+require 'reek/cli/report/formatter'
+require 'reek/cli/report/heading_formatter'
+
+module Reek
+  module Cli
+    #
+    # Interprets the options set from the command line
+    #
+    class OptionInterpreter
+      include Cli::Input
+
+      extend Forwardable
+
+      def_delegators :@options, :smells_to_detect
+
+      def initialize(options)
+        @options = options
+        @argv = @options.argv
+      end
+
+      def reporter
+        @reporter ||=
+          report_class.new(
+            warning_formatter: warning_formatter,
+            report_formatter: Report::Formatter,
+            sort_by_issue_count: sort_by_issue_count,
+            heading_formatter: heading_formatter)
+      end
+
+      def report_class
+        case @options.report_format
+        when :yaml
+          Report::YamlReport
+        when :html
+          Report::HtmlReport
+        else # :text
+          Report::TextReport
+        end
+      end
+
+      def warning_formatter
+        klass = if @options.show_links
+                  Report::UltraVerboseWarningFormatter
```

Comment:
Then we probably should rename the Formatter as well to "ShowLinksWarningFormatter" - what do you think?